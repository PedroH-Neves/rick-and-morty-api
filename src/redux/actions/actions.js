export const FETCH_CHARACTERS_REQUEST = 'FETCH_CHARACTERS_REQUEST';
export const FETCH_CHARACTERS_SUCCESS = 'FETCH_CHARACTERS_SUCCESS';
export const FETCH_CHARACTERS_FAILURE = 'FETCH_CHARACTERS_FAILURE';
export const ADD_FILTER = 'ADD_FILTER';
export const REMOVE_FILTER = 'REMOVE_FILTER';


export const fetchCharactersRequest = () => ({type:FETCH_CHARACTERS_REQUEST});
export const fetchCharactersSuccess = characters => ({type:FETCH_CHARACTERS_SUCCESS, payload: characters});
export const fetchCharactersFailure = error => ({type: FETCH_CHARACTERS_FAILURE, payload: error});
export const addFilter = filter => ({ type: ADD_FILTER, payload: filter });
export const removeFilter = filter => ({ type: REMOVE_FILTER, payload: filter });
