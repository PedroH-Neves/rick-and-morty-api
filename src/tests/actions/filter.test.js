import configureMockStore from 'redux-mock-store';
import { addFilter, removeFilter } from '../../redux/actions/actions';

const mockStore = configureMockStore();
const store = mockStore({});

describe('actions creation', () => {
  it('should add a filter', () => {
    store.dispatch(addFilter(
      {
        status: 'Alive',
      },
    ));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should remove a filter', () => {
    store.dispatch(removeFilter(
      {
        status: ['Alive', 'Dead'],
      },
    ));
    expect(store.getActions()).toMatchSnapshot();
  });
});
