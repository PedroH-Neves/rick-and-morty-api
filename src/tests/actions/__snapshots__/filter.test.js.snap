// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions creation should add a filter 1`] = `
Array [
  Object {
    "payload": Object {
      "status": "Alive",
    },
    "type": "ADD_FILTER",
  },
]
`;

exports[`actions creation should remove a filter 1`] = `
Array [
  Object {
    "payload": Object {
      "status": "Alive",
    },
    "type": "ADD_FILTER",
  },
  Object {
    "payload": Object {
      "status": Array [
        "Alive",
        "Dead",
      ],
    },
    "type": "REMOVE_FILTER",
  },
]
`;
